File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/works_approval/SearchWorksApprovalResultDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.works_approval;

import java.io.Serializable;

/**
 * 作業承認検索結果Dto
 */
public class SearchWorksApprovalResultDto implements Serializable { // NOPMD DataClass

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 承認作業住所検索結果Dto */
    private SearchApprovalAddressResultDto resultDtoAddress = new SearchApprovalAddressResultDto();

    /** 承認作業職業検索結果Dto */
    private SearchApprovalShokugyouResultDto resultDtoShokugyou = new SearchApprovalShokugyouResultDto();

    /**
     * 承認作業住所検索結果Dtoを取得する
     *
     * @return 承認作業住所検索結果Dto
     */
    public SearchApprovalAddressResultDto getResultDtoAddress() {
        return resultDtoAddress;
    }

    /**
     * 承認作業住所検索結果Dtoを設定する
     *
     * @param resultDtoAddress 承認作業住所検索結果Dto
     */
    public void setResultDtoAddress(final SearchApprovalAddressResultDto resultDtoAddress) {
        this.resultDtoAddress = resultDtoAddress;
    }

    /**
     * 承認作業職業検索結果Dtoを取得する
     *
     * @return 承認作業職業検索結果Dto
     */
    public SearchApprovalShokugyouResultDto getResultDtoShokugyou() {
        return resultDtoShokugyou;
    }

    /**
     * 承認作業職業検索結果Dtoを設定する
     *
     * @param resultDtoShokugyou 承認作業職業検索結果Dto
     */
    public void setResultDtoShokugyou(final SearchApprovalShokugyouResultDto resultDtoShokugyou) {
        this.resultDtoShokugyou = resultDtoShokugyou;
    }

}

File: app/front/src/components/main/dto/works_approval/searchWorksApprovalResultDto.ts
import { SearchApprovalAddressResultDto, type SearchApprovalAddressResultDtoInterface } from "./searchApprovalAddressResultDto"
import { SearchApprovalShokugyouResultDto, type SearchApprovalShokugyouResultDtoInterface } from "./searchApprovalShokugyouResultDto"

interface SearchWorksApprovalResultDtoInterface {

    /** 承認作業住所検索結果Dto */
    resultDtoAddress: SearchApprovalAddressResultDtoInterface;

    /** 承認作業職業検索結果Dto */
    resultDtoShokugyou: SearchApprovalShokugyouResultDtoInterface;

}

class SearchWorksApprovalResultDto {

    /** 承認作業住所検索結果Dto */
    resultDtoAddress: SearchApprovalAddressResultDtoInterface;

    /** 承認作業職業検索結果Dto */
    resultDtoShokugyou: SearchApprovalShokugyouResultDtoInterface;

    constructor() {
        this.resultDtoAddress = new SearchApprovalAddressResultDto();
        this.resultDtoShokugyou = new SearchApprovalShokugyouResultDto();
    }

}

export { type SearchWorksApprovalResultDtoInterface, SearchWorksApprovalResultDto }